### PR TITLE
CT-203-NT-get history flow optimization

### PR DESCRIPTION
**Pull request by Nadin-a**
added the logic for get history api ( + encryption / decryption ), added pagination and timing, added typescript to project

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,10 +2,14 @@
...
   "version": "1.0.0",
   "description": "",
-  "main": "src/index.js",
+  "main": "dist/src/index.js",
   "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1",
-    "docs": "swagger-jsdoc -d swaggerfile.js"
+    "docs": "swagger-jsdoc -d swaggerfile.js",
+    "build-bitcore-lib": "browserify -r bitcore-lib -s bitcore-lib -o src/assets/libs/bitcore-lib.js",
+    "build-zcash-bitcore-lib": "browserify -r zcash-bitcore-lib -s zcash-bitcore-lib -o src/assets/libs/zcash-bitcore-lib.js",
```
Comment (Isahara86) [inline package.json]: no sence to browserify "bitcore-lib" & "zcash-bitcore-lib" they was created for nodejs

Comment (Nadin-a) [reply package.json]: fixed

File: src/history.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    } else {
+        return res.json({
+            status: 'error',
+            result: [],
+        });
+    }
+});
+
+function getWallet(coinType: string, options: WalletCreateOptionsInterface) : WalletLike {
+    let wallet: WalletLike =  new WalletBtc(options);
```
Comment (Isahara86) [inline src/history.ts]: let wallet: WalletLike =  new WalletBtc(options);
should be 
let wallet: WalletLike;

File: src/history.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+        return res.json({
+            status: 'success',
+            result: result,
+            time: timeResult.time,
+        });
+    } else {
+        return res.json({
+            status: 'error',
+            result: [],
```
Comment (Isahara86) [inline src/history.ts]: error should be explicit. 
Example: "unsupported coinType"

File: package.json [hunk 4]
```diff
@@ -24,9 +26,23 @@
   },
   "homepage": "https://github.com/agilie/electrum-proxy-middleware#readme",
   "dependencies": {
-    "express-async-router": "^0.1.15"
+    "@dashevo/dashcore-lib": "^0.17.9",
+    "@types/lodash": "^4.14.138",
+    "bitcore-lib": "^8.3.4",
+    "ethers": "^4.0.36",
```
Comment (Isahara86) [inline package.json]: this lib is also useless.

File: package.json [hunk 5]
```diff
@@ -24,9 +26,23 @@
...
+    "execution-time": "^1.3.0",
+    "express-async-router": "^0.1.15",
+    "js-sha256": "^0.9.0",
+    "rxjs": "^6.5.3",
+    "service": "^0.1.4",
+    "tslint": "^5.19.0",
```
Comment (Isahara86) [inline package.json]: "tslint": "^5.19.0", - is a dev dependency

File: src/errors/electrum-config-validator.ts [hunk 6]
```diff
@@ -0,0 +1,62 @@
+import {CoinType} from '../service/wallet/types/coin.type';
+
+export class ElectrumConfigValidator {
```
Comment (Isahara86) [inline src/errors/electrum-config-validator.ts]: please consider to use class-validator package instead

File: src/index.ts [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+router.use('/server', require('./server'));
+router.use('/transaction', require('./transaction'));
+router.use('/address', require('./address'));
+router.use('/block', require('./block'));
+router.use('/blockchain', require('./blockchain'));
+router.use('/scripthash', require('./scripthash'));
+router.use('/mempool', require('./mempool'));
+router.use('/history', require('./history'));
+
+module.exports = router;
```
Comment (Isahara86) [inline src/index.ts]: use TS export not JS

File: src/history.ts [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+function validateCoinType(coinType: CoinType, res: any) {
+    if(!coinType) {
+        return res.json({
+            status: 'error',
+            result: 'Coin type is missing'
+        });
+    }
+}
+
+module.exports = router;
```
Comment (Isahara86) [inline src/history.ts]: use TS export not JS

**Pull request merged**